packages/kit/src/pages.ts

```diff
@@ -11,3 +13,28 @@ export function extendPages (cb: NuxtHooks['pages:extend']) {
     nuxt.hook('pages:extend', cb)
   }
 }
+
+export interface ExtendRouteRulesOptions {
+  /**
+   * Override route rule config
+   *
+   * @default false
+   */
+  override?: boolean
+}
+
+export function extendRouteRules (route: string, rule: NitroRouteConfig, options: ExtendRouteRulesOptions = {}) {
+  const nuxt = useNuxt()
+  if (!nuxt.options.routeRules) {
+    nuxt.options.routeRules = {}
+  }
+  nuxt.options.routeRules[route] = options.override
+    ? defu(rule, nuxt.options.routeRules[route])
+    : defu(nuxt.options.routeRules[route], rule)
+  if (!nuxt.options.nitro.routeRules) {
+    nuxt.options.nitro.routeRules = {}
+  }
+  nuxt.options.nitro.routeRules[route] = options.override
+    ? defu(rule, nuxt.options.nitro.routeRules[route])
+    : defu(nuxt.options.nitro.routeRules[route], rule)
```

Comment:
feels like we could optimise slightly:

```suggestion
  for (const options of [nuxt.options, nuxt.options.nitro]) {
	  if (!options.routeRules) {
	    options.routeRules = {}
	  }
	  options.routeRules[route] = options.override
	    ? defu(rule, options.routeRules[route])
	    : defu(options.routeRules[route], rule)
  }
```